config/config.php

```diff
@@ -17,4 +30,32 @@
         'name' => '/^(route-usage\.|nova\.)/',
         'uri' => ''
     ],
+
+    /*
+    |--------------------------------------------------------------------------
+    | Environment
+    |--------------------------------------------------------------------------
+    |
+    | If you don't want to use this package in production env
+    | at all, you can restrict that using this option
+    | rather than by using a middleware.
+    |
+    */
+
+    'allowed_environments' => ['local', 'staging', 'testing'],
```

Comment:
This goes against the point of this package. It's mostly meant to run on production to know what is used **in real life**. And if it runs in production, it's probably best to always run it.

Reply:
Okay we can remove this but for my case i don't want to run in production but in our staging site rather

Reply:
Yes, you're right we need an easy way to turn off this package.

I'd rather not do it in a config entry based on environment. I would like a method or closure that you can define yourself. Not sure exactly yet what would be the best way.

What do you think of overriding the ServiceProvider and defining something like

```php
class RouteUsageServiceProvide extend PackageServiceProvider
{
    public function shouldListen()
    {
        return ! app()->environment('production');
    }
}
```

Maybe people don't like overriding service provider and prefer defining a closure like in `Telescope::auth(..)` 🤔

Reply:
Yeah most of the package users will likely want it in `config` files